spiped/dispatch.c

```diff
@@ -250,6 +302,16 @@ void
 dispatch_shutdown(void * dispatch_cookie)
 {
 	struct accept_state * A = dispatch_cookie;
+	struct conn_list_node * ptr;
+	struct conn_list_node * ptr_next;
+
+	/* Shutdown any open connections. */
+	ptr = A->conn_cookies;
+	while (ptr != NULL) {
+		ptr_next = ptr->next;
+		proto_conn_drop(ptr->conn_cookie);
+		ptr = ptr_next;
```

Comment:
`proto_conn_drop` calls back to `callback_conndied()`, which frees the linked list nodes.  I could add a comment to that effect?  Or would you rather have a different code design?